File: 56-merge-intervals/merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """

        intervals.sort(key=lambda x: x[0])  # Sort intervals by start time

        merged = []
        for interval in intervals:
          if not merged or merged[-1][1] < interval[0]:
            merged.append(interval)
          else:
            merged[-1][1] = max(merged[-1][1], interval[1])

        return merged

File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int n = intervals.size();
        vector<vector<int>>ans;

        for (int i = 0; i < n; i++){
            if (ans.size() == 0 || ans[ans.size()-1].back() < intervals[i][0]){
                ans.push_back(intervals[i]);
            }
            else {
                ans[ans.size()-1].back() = max(ans[ans.size()-1].back(), intervals[i][1]);
            }
        }

        return ans;
    }
};